src/receivers/otlp_http.rs

```diff
@@ -443,22 +506,28 @@ where
         return response_4xx(StatusCode::BAD_REQUEST);
     }
 
-    let metrics_req = decoded.unwrap();
-    //let x = ExpReq::otlp_into(metrics_req);
+    let req = decoded.unwrap();
 
     let mut rb = Response::builder();
     rb.headers_mut()
         .unwrap()
         .insert(CONTENT_TYPE, HeaderValue::from_str(PROTOBUF_CT).unwrap());
 
-    match output.send(ExpReq::otlp_into(metrics_req)).await {
+    let otlp_payload = ExpReq::otlp_into(req);
+    let count = BatchSizer::size_of(&otlp_payload);
+
+    match output.send(otlp_payload).await {
         Ok(_) => {
             // No partial success at the moment
-            let body = Full::new(metrics_ok_resp.clone());
+            let body = Full::new(ok_resp.clone());
+            accepted_counter.add(count as u64, &[KeyValue::new("protocol", "http")]);
```

Comment:
Any benefit to memoize the KeyValue here or should that be optimized by the compiler?

Reply:
I haven't looked at the assembly but I don't think it'll hurt either way to store it and clone